src/http/server.cr

```diff
@@ -138,8 +138,8 @@ class HTTP::Server
     end
   end
 
-  def bind
-    @server ||= TCPServer.new(@host, @port)
+  def bind(reuse_port = false)
```

Comment:
A few things:
- `host` and `port` aren't properties, why should `reuse_port` be?
- `reuse_port`, as `host` and `port`, only applies *before* binding the socket;
- stopping and restarting HTTP::Server... will that happen? I'm not sure this is even possible (as of now).

Maybe an argument to `#new` would be better, just like `TCPServer#new`.